lib/solarwinds_apm/support/otlp_endpoint.rb

```diff
@@ -0,0 +1,203 @@
+    end
+
+    def config_otlp_endpoint
+      config_service_name
+      config_token
+      %w[TRACES METRICS LOGS].each { |data_type| configure_otlp_endpoint(data_type) }
+    end
+
+    def config_service_name
+      resource_attributes = ENV['OTEL_RESOURCE_ATTRIBUTES'].to_s.split(',').each_with_object({}) do |resource, hash|
+        key, value = resource.split('=')
+        hash[key] = value
+      end
+
+      @service_name = if @lambda_env
+                        ENV['OTEL_SERVICE_NAME'] || ENV['AWS_LAMBDA_FUNCTION_NAME'] || resource_attributes['service.name']
+                      else
+                        ENV['OTEL_SERVICE_NAME'] || resource_attributes['service.name'] || @service_name || 'None'
+                      end
+
+      ENV['OTEL_SERVICE_NAME'] = @service_name
+    end
+
+    def mask_token(token)
+      token = token.to_s
+      return '*' * token.length if token.length <= 4
+
+      "#{token[0, 2]}#{'*' * (token.length - 4)}#{token[-2, 2]}"
+    end
+
+    def config_token(data_type)
+      agent_enable = true
+
+      if @localhost && !@lambda_env
+        # for localhost, only valid SW_APM_SERVICE_KEY can make agent enable (except inside lambda case)
+        @agent_enable = false unless valid?(ENV['SW_APM_SERVICE_KEY'])
+        return
+      end
+
+      data_type_upper = data_type.upcase
+      data_type = data_type.downcase
+
+      if @lambda_env
+        # for case 10 and 11
+        agent_enable = !ENV['SW_APM_API_TOKEN'].nil?
+      else
+        token_source = if ENV["OTEL_EXPORTER_OTLP_#{data_type_upper}_HEADERS"]
+                       "#{data_type}_token"
+                     elsif ENV['OTEL_EXPORTER_OTLP_HEADERS']
+                       'general_token'
+                     elsif ENV['SW_APM_SERVICE_KEY']
+                       'service_key'
+                     else
+                       'invalid'
+                     end
+
+        SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] token source: #{token_source}" }
+
+        if token_source == 'service_key'
+          if valid?(ENV['SW_APM_SERVICE_KEY'])
+            @token, @service_name = ENV['SW_APM_SERVICE_KEY'].to_s.split(':')
+          else
+            token_source = 'invalid'
+            SolarWindsAPM.logger.warn { "SW_APM_SERVICE_KEY is invalid: #{mask_token(ENV['SW_APM_SERVICE_KEY'])}" }
+          end
+
+          ENV['OTEL_EXPORTER_OTLP_HEADERS'] = "authorization=Bearer #{@token}"
+        end
+
+        agent_enable = token_source != 'invalid'
+      end
+
+      @agent_enable = agent_enable
+    end
+
+    def valid?(_service_key)
+      # servicekey checker also works on service name, may need to remove that part
+      service_key_checker = SolarWindsAPM::ServiceKeyChecker.new('ssl', false)
+      service_key_name = service_key_checker.read_and_validate_service_key
+      service_key_name != ''
+    end
+
+    def determine_lambda_env
```

Comment:
can the existing [determine_lambda](https://github.com/solarwinds/apm-ruby/blob/0595edffca4325f4c2d5b2bfbe01d9185611c088/lib/solarwinds_apm/oboe_init_options.rb#L209) method be used or made shared by both these modules?

Reply:
Updated.